src/core/video_channel.cpp

```diff
@@ -78,6 +78,19 @@ struct video_channel::impl final
     std::atomic<bool> abort_request_{false};
     std::thread       thread_;
 
+    std::function<void(int, core::draw_frame)> routesCb = [&](int layer, core::draw_frame frame) {
+        std::lock_guard<std::mutex> lock(routes_mutex_);
+        for (auto& r : routes_) {
```

Comment:
Im not keen on how this will iterate through all of routes_ once for every layer

Reply:
As discussed, the change to iterating routes for each layer is required to achieve the zero-delay target. I have optimised the loop and with the likely number of routes per channel this should not be a problem.